src/model_config_tests/models/accessesm1p5.py

```diff
@@ -74,14 +85,33 @@ def extract_checksums(
         self, output_directory: Path = None, schema_version: str = None
     ) -> dict[str, Any]:
         """Parse output file and create checksum using defined schema"""
-        if output_directory:
-            output_filename = output_directory / self.output_filename
-        else:
+
+        submodels = {
+            submodel["model"]: submodel["name"]
+            for submodel in self.experiment.config["submodels"]
+        }
+
+        # Extract checksums from output, preferentially using mom5
+        submodel_extract_checksums = None
+        if "mom" in submodels:
             output_filename = self.output_file
+            submodel_extract_checksums = mom5_extract_checksums
+        elif "um" in submodels:
+            # UM output is stored in submodel ouptut sub-directory
+            output_filename = self.output_0 / submodels["um"] / "atm.fort6.pe0"
+            submodel_extract_checksums = um7_extract_norms
+
+        assert submodel_extract_checksums is not None, (
+            "Failed to find suitable submodel for checksum extraction."
+            "Required 'mom' or 'um' submodels not present in config.yaml "
+        )
+
+        if output_directory is not None:
+            output_filename = output_directory / output_filename
```

Comment:
Output directory is an option for when we want to check an output directory that isn't `output000`, e.g. the repro tests that runs the experiment twice sets `output_directory` to `/path/to/archive/output001` for the second model run. So we need the `output_filename` to be filename or a relative path? At the moment it is an absolute path in the output000 directory. Could also change the variable name for the absolute/full path to file from `output_filename`?
```suggestion
        if output_directory is not None:
            output_filepath = output_directory / output_filename
        else:
            output_filepath = self.output_0 / output_filename
```
Sorry I should've caught this on the first review!

Reply:
Thanks for catching this, and no need to apologise – I should have caught it while writing it!

I've added this in a72956c.

I've moved the section which sets the filename up into the initialisation. I noticed the reproducibility tests check that `self.output_file` exists before the checksums are extracted. In the earlier commits in this PR, self.output_file didn't necessarily match the file that the checksums were being extracted from, and so it was checking the wrong file. I've added some changes which should hopefully fix this!

Let me know if you have any suggestions!

Reply:
Good catch! Yeah that check was a way to check if the model ended up running.. I'm currently looking into parsing out the exit code of jobs in a different issue to check if a payu exited with errors. But anyway still good to check that the file needed for checksums exists